### PR TITLE
update yaml branch

### PR DESCRIPTION
**Pull request by ggilmore**
- updates all docker images
- uses stateful sets from https://github.com/sourcegraph/deploy-sourcegraph/pull/65
- apparently more prettier formatting 

The scripts haven't been updated, only the base deployment

### CODE REVIEW

File: base/backend.Service.yaml [hunk 1]
```diff
@@ -2,18 +2,18 @@ apiVersion: v1
 kind: Service
 metadata:
   annotations:
-    description: Dummy service that prevents backend pods from being scheduled on
+    description:
+      Dummy service that prevents backend pods from being scheduled on
       the same node if possible.
   labels:
-    deploy: sourcegraph
```
Comment (nicksnyder) [inline base/backend.Service.yaml]: all resources should have this

File: base/frontend/sourcegraph-frontend.Deployment.yaml [hunk 2]
```diff
@@ -21,56 +21,60 @@ spec:
...
-          requests:
-            cpu: "2"
-            memory: 1G
-        volumeMounts:
-        - mountPath: /etc/sourcegraph
-          name: sg-config
+        - args:
+            - serve
+          env:
+            - name: DEPLOY_TYPE
```
Comment (nicksnyder) [inline base/frontend/sourcegraph-frontend.Deployment.yaml]: shouldn't be necessary:
https://sourcegraph.sgdev.org/github.com/sourcegraph/sourcegraph/-/blob/pkg/conf/computed.go#L126:10

File: base/indexer/indexer.Deployment.yaml [hunk 3]
```diff
@@ -21,38 +21,40 @@ spec:
...
-            cpu: "1"
-            memory: 1G
-          requests:
-            cpu: 100m
-            memory: 250M
-        volumeMounts:
-        - mountPath: /etc/sourcegraph
-          name: sg-config
+        - env:
+            - name: LSP_PROXY
```
Comment (nicksnyder) [inline base/indexer/indexer.Deployment.yaml]: this shouldn't be necessary

File: base/frontend/sourcegraph-frontend.Deployment.yaml [hunk 4]
```diff
@@ -21,56 +21,60 @@ spec:
...
+              value: datacenter
+            - name: LSP_PROXY
```
Comment (nicksnyder) [inline base/frontend/sourcegraph-frontend.Deployment.yaml]: shouldn't be necessary
https://sourcegraph.sgdev.org/github.com/sourcegraph/sourcegraph/-/blob/xlang/client.go#L28:21

**Comment by ggilmore** [comment]
@nicksnyder PTAL, I reformatted everything with [a small program that I wrote](https://github.com/ggilmore/k8sfmt) - it should be easier to review now

**Pull request merged**